### ISSUE TITLE
distributed.utils - ERROR - Worker already exists

### ISSUE THREAD

**Issue by Sutyke** [dask/distributed#2540]
I would like to ask if there is any solution to the issue below. at the moment dask can not be used for scaling tpot as when running after while there is cumming error message and dask freeze. 

For more info please  see the link documentation below: 
https://examples.dask.org/machine-learning/tpot.html

In the section "Using Dask" when tp.fit(X_train, y_train) I'm getting the same error as below:

distributed.utils - ERROR - Worker already exists tcp://127.0.0.1:46060
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/distributed/utils.py", line 648, in log_errors
    yield
  File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/distributed/scheduler.py", line 1278, in add_worker
    raise ValueError("Worker already exists %s" % address)
ValueError: Worker already exists tcp://127.0.0.1:46060
distributed.core - ERROR - Worker already exists tcp://127.0.0.1:46060
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/distributed/core.py", line 346, in handle_comm
    result = yield result
  File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/tornado/gen.py", line 326, in wrapper
    yielded = next(result)
  File "/home/travis/miniconda/envs/test/lib/python3.6/site-packages/distributed/scheduler.py", line 1278, in add_worker
    raise ValueError("Worker already exists %s" % address)
ValueError: Worker already exists tcp://127.0.0.1:46060

Could you please help if there is any possibility to run dask with tpot at the moment?

Thanks for advice

**Comment by TomAugspurger** [comment]
Moved to distributed.

@Sutyke can you expand a bit on what you're doing / seeing? Are you running the example from 
https://examples.dask.org/machine-learning/tpot.html exactly as is? That error message is strange, since there shouldn't be kind of autoscaling there that would trigger adding workers.

**Comment by Sutyke** [comment]
I'm running TPOT classification on the the matrix 1700x400 using dask, the problem happens on both local or remote cluster. TPOT works excellent when  I don't use dask.

When I use dask , workers are getting killed even there is plenty of memory( 350GB per worker and I'm using 2-4 workers) . I'm getting this message  when new worker is not created. I can recreate worker manually, but once worker is killed, dask starts to freeze.

I spent 2 weeks with testing. Error is not consistent , sometimes it's just freeze even worker is recreate automatically by dask. Dask runs fine tool first worker is killed.

I can upload the code and data I'm using if you would like to run but as you seem on the dask webpage others are getting the same error message and it's documented in the docs.

Bottom line is that dask is excellent tool but can not be used on TPOT as I'm getting faster results with running without dask.

Let me know if I can provide more information, dask is excellent tool just if it's not robust it's useful as proved in my case.

Rg
Dan

**Comment by TomAugspurger** [comment]
Closing this as a duplicate of https://github.com/dask/dask/issues/4769. If resolving that doesn't also resolve this then we'll reopen.
